src/Build/BackEnd/Components/Logging/LoggingService.cs

```diff
@@ -1666,8 +1666,8 @@ private void RouteBuildEvent(object loggingEvent)
                 }
             }
 
-            // If this is BuildCheck-ed build - add the warnings promotability/demotability to the service
-            if (buildEventArgs is ProjectStartedEventArgs projectStartedEvent && this._componentHost.BuildParameters.IsBuildCheckEnabled)
+            // Respect warning-promotion properties from the remote project
+            if (buildEventArgs is ProjectStartedEventArgs projectStartedEvent)
```

Comment:
It'll need the worker node side adjust as well:

https://github.com/dotnet/msbuild/blob/5e119d75b47cbf7f35cdcb0690f2bbb4f6ef2288/src/Build/BackEnd/Components/RequestBuilder/RequestBuilder.cs#L1282


Hopefuly it shouldn't have significant perf impact - but I haven't test that as it was limited to BuildCheck that's perf hit anyways.

Reply:
I don't think I understand why that needs to change. This does fix the basic issue in #10874.

Reply:
If receiving of `ProjectStartedEventArgs.Warnings[Not]As[Errors|Messages]` is wanted to be active regardless of BuildCheck being on - then the population of those properties should not be conditioned on that either. So the `buildCheckManager != null &&` should be removed in that code.

It worked because that conditioning was buggy - it should have conditioned based on `BuildParameters.IsBuildCheckEnabled` - as `buildCheckManager` has the "null object" value if BuildCheck is disabled - so the current condition is misleading. On top of that it causes skipping of the code during restore - which is the stage where we actualy want that to transfer the information.